File: Data Structures & Algorithms/is-palindrome/submission-0.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string str="";
        int left,right;

        for(char c:s){
            if(isalnum(c)){
                str += tolower(c);
            }
        }
        left=0;
        right=str.size()-1;
        while(left<right){
            if(str[right]!=str[left]){
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
};

File: Data Structures & Algorithms/same-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    boolean a=true;
    public boolean isSameTree(TreeNode p, TreeNode q) {

        if(p==null||q==null){
            if(p!=q){
                a=false;
                return false;
            }else{
                return true;
            }
        }
        if(p.val==q.val&&a==true){
            isSameTree(p.left,q.left);
            isSameTree(p.right,q.right);
        }else{
            a=false;
            return false;
        }

        return a;    
    }
}
